src/SFA.DAS.ApprenticeFeedback.Api/AppStart/AddDatabaseRegistrations.cs

```diff
@@ -23,7 +23,12 @@ public static void AddDatabaseRegistration(this IServiceCollection services, ICo
             }
             else
             {
-                services.AddSingleton(new AzureServiceTokenProvider());
+                services.AddSingleton(new ChainedTokenCredential(
+                    new ManagedIdentityCredential(),
```

Comment:
Did you do any investigation about what these additional singletons where needed for other than the ChainedTokenCredential?

Reply:
From what I understand, `ChainedTokenCredential` accepts any number of objects derived from `TokenCredential`, and tries each of them until authentication is successful. This is the combination that I've found to work, but others might work as well